packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/LabelBoundsProvider.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams;
+
+import java.awt.Font;
+import java.awt.GraphicsEnvironment;
+import java.awt.font.FontRenderContext;
+import java.awt.geom.AffineTransform;
+import java.util.Comparator;
+import java.util.stream.Stream;
+
+/**
+ * Provides the TextBounds of a given text applied to a LabelStyle.
+ * @author mcharfadi
+ */
+public class LabelBoundsProvider {
+    private static final String DEFAULT_LABEL_FONT_NAME = "Arial";
+    private static final String FALLBACK_LABEL_FONT_NAME = "Liberation Sans";
+    private static final AffineTransform AFFINE_TRANSFORM = new AffineTransform();
+    private static final FontRenderContext FONT_RENDER_CONTEXT = new FontRenderContext(AFFINE_TRANSFORM, true, true);
+    private static String fontName;
+
+    public static Size getLabelBounds(LabelStyle labelStyle, String text) {
+        Font font = getFont(labelStyle);
+
+        String maxGlyphHeight = "Éç";
+        Double maxFontHeight = font.getStringBounds(maxGlyphHeight, FONT_RENDER_CONTEXT).getHeight();
+        Double totalHeight = maxFontHeight * text.lines().count();
+
+        String longestLine = text.lines().max(Comparator.comparingInt(String::length)).get();
+        Double maxWidthValue = font.getStringBounds(longestLine, FONT_RENDER_CONTEXT).getWidth();
+
+        return Size.of(Math.round(maxWidthValue), Math.round(totalHeight));
+    }
+
+    private static int round(double d) {
```

Comment:
It's only used to `round(nbLigne)`, which can only be positive.
So `dAbs == d`, and this is equivalent to `((int) d) + 1`, which return 2 for d=1.0, which is probably wrong.